### PR TITLE
feat: add isNonEmptyArray

### PR DESCRIPTION
**Pull request by guillaumearm**
Ref #279

### CODE REVIEW

File: test/isNonEmptyArray.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import * as RA from '../src/index';
+import eq from './shared/eq';
+
```
Comment (char0n) [inline test/isNonEmptyArray.js]: insert one new empty line pls

File: src/isNonEmptyArray.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { both } from 'ramda';
+
+import isNotEmpty from './isNotEmpty';
+import isArray from './isArray';
+
```
Comment (char0n) [inline src/isNonEmptyArray.js]: insert one new empty line pls

File: src/index.d.ts [hunk 3]
```diff
@@ -53,6 +53,11 @@ declare namespace RamdaAdjunct {
          */
         isNotArray(val: any): boolean;
 
+        /**
+         * Checks if input value is a non empty `Array`.
+         */
+        isNonEmptyArray(val: any): val is Array<any>;
```
Comment (char0n) [inline src/index.d.ts]: @BjornMelgaard I'm not sure about this. Can you advice ? Does this signature make sense in the context of empty array ?

File: src/isNonEmptyArray.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * RA.isNonEmptyArray([42]); // => true
+ * RA.isNonEmptyArray([]); // => false
+ * RA.isNonEmptyArray({}); // => false
+ * RA.isNonEmptyArray(null); // => false
+ * RA.isNonEmptyArray(undefined); // => false
+ * RA.isNonEmptyArray(42); // => false
+ * RA.isNonEmptyArray('42'); // => false
+ */
+const isNonEmptyArray = both(isNotEmpty, isArray);
```
Comment (char0n) [inline src/isNonEmptyArray.js]: I'd prefer one of the following compositions:
```js
const isNonEmptyArray = both(isArray, isNotEmpty);
```
or
```js
const isNonEmptyArray = both(isArray, complement(isEmptyArray)); 
```

File: src/index.d.ts (hunk 3 above)
Comment (BjornMelgaard) [reply src/index.d.ts]: @char0n I think yes, there is no way we can express that array is non-empty in types

Comment (char0n) [reply src/index.d.ts]: Wouldn't be better to use `boolean` instead ?

Comment (BjornMelgaard) [reply src/index.d.ts]: Look
https://www.typescriptlang.org/play/index.html#src=declare%20function%20isNonEmptyArray(val%3A%20any)%3A%20val%20is%20Array%3Cany%3E%3B%0D%0A%0D%0Afunction%20createName(name%3A%20any)%20%7B%0D%0A%20%20%20%20if%20(isNonEmptyArray(name))%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20name.join()%0D%0A%20%20%20%20%7D%0D%0A%20%20%20%20else%20%7B%0D%0A%20%20%20%20%20%20%20%20return%20name.join(%22%20%22)%3B%0D%0A%20%20%20%20%7D%0D%0A%7D

when `isNonEmptyArray` returns `false` array methods still can be applied on `name`

so `val is Array<any>` if fine )

**Pull request merged**